target_snowflake/db_sync.py

```diff
@@ -762,6 +762,18 @@ def update_columns(self):
 
         columns_dict = {column['COLUMN_NAME'].upper(): column for column in columns}
 
+        # check if column data type matches between exporting data and target table
+        columns_with_type_mismatch = [
+            name.upper()
+            for (name, properties_schema) in self.flatten_schema.items()
+            if name.upper() in columns_dict and
+               columns_dict[name.upper()]['DATA_TYPE'].upper() != column_type(properties_schema).upper() and
+               not (columns_dict[name.upper()]['DATA_TYPE'].upper() in ['DATE', 'TIMESTAMP', 'TIMESTAMP_TZ', 'TIMESTAMP_NTZ', 'TIMESTAMP_LTZ'] and column_type(properties_schema).upper() == 'TIMESTAMP_NTZ')
```

Comment:
Snowflake has DATE, TIMESTAMP, TIMESTAMP_TZ, TIMESTAMP_NTZ, TIMESTAMP_LTZ for date types and target uses TIMESTAMP_NTZ type for exporting date column. Tested and verified that dates exported using TIMESTAMP_NTZ fits into all date type in Snowflake.